### ISSUE TITLE
compression defaults

### ISSUE THREAD

**Issue by alonpr** [confluentinc/ksql#9207]
In:
[compression.type](https://docs.ksqldb.io/en/latest/reference/server-configuration/#compressiontype)
and
[ksql.streams.producer.compression.type](https://docs.ksqldb.io/en/latest/reference/server-configuration/#ksqlstreamsproducercompressiontype)
it says: _"The default is snappy"_ but the default seems to be 'none' (unless explicitly set in server configuration file or set for the session).

**Comment by JimGalasyn** [comment]
@alonpr Thank you for the heads up! @jnh5y Just to confirm: the default, as seen by ksqlDB, is "none", and not "snappy"?

**Comment by jnh5y** [comment]
Yes, I do believe that's correct.  I do not see anything in the ksql codebase which is overriding the default.  

Further, (I think that) the AK default is set here: https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java#L352

If CP sets `compression.type` and/or `ksql.streams.producer.compression.type`, that certainly makes sense.  

@colinhicks should we fix this as a docs bug or should we fix the code so that `snappy` is used by default?  (Admittedly, we could backport a docs fix and fix the config going forward.)

**Comment by JimGalasyn** [comment]
@jnh5y @colinhicks I have a PR up if we decide we want to fix the docs: https://github.com/confluentinc/ksql/pull/9208.

**Comment by colinhicks** [comment]
There is an existing ticket to unify these configs so that there is less confusion moving forward: https://github.com/confluentinc/ksql/issues/7351. I think we should not change any defaults until we tackle that.

**Comment by colinhicks** [comment]
I think another source of the confusion is that Confluent Cloud KSQL does in fact configure `snappy` (for both configs here), while other KSQL distributions do not specify the compression.

**Comment by JimGalasyn** [comment]
@colinhicks kk cool, maybe I'll just leave that PR for now.
